### PR TITLE
done

### PR DESCRIPTION
**Pull request by not4und**

### CODE REVIEW

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    public String getRandomColor() {
+        return Color.values()[new Random().nextInt(Color.values().length)].name();
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: make a global Random field and use it here

File: src/main/java/core/basesyntax/Figure.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                        .append(", area: ")
+                        .append(String.format("%.1f", this.area()))
+                        .append(" sq.units, ")
+                        .append(this.getSizes())
+                        .append(", color: ")
+                        .append(color.toLowerCase())
+        );
+    }
+
+    public StringBuilder getSizes() {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Figure.java:27]: remove this method you don't need it, if you want to use StringBulider use it directly in every figure

File: src/main/java/core/basesyntax/Figure.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    private final String color;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public abstract double area();
+
+    @Override
+    public void draw() {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Figure.java:13]: you should've implement in this method in each figure, not here

File: src/main/java/core/basesyntax/Figure.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable {
+    private final String color;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public abstract double area();
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/Figure.java:10]: create a separate interface for this method + think about better naming, names of methods should start with a verb not noun

File: src/main/java/core/basesyntax/FigureForm.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public enum FigureForm {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/FigureForm.java:3]:
```suggestion
public enum FigureType {
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_SIZE_LENGTH = 25;
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/FigureSupplier.java:6]:
```suggestion
    private static final int MAX_RANDOM_NUMBER = 25;
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        FigureForm figureForm = FigureForm.values()[random.nextInt(FigureForm.values().length)];
+
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/FigureSupplier.java:12]: remove empty line

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/FigureSupplier.java:5]: I don't see this method here
Screenshot 2022-08-18 at 09 40 02

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+                        random.nextInt(MAX_SIZE_LENGTH)
+                );
+            case ISOSCELES_TRAPEZOID:
+            default:
+                return new IsoscelesTrapezoid(
+                        colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIZE_LENGTH),
+                        random.nextInt(MAX_SIZE_LENGTH),
+                        random.nextInt(MAX_SIZE_LENGTH)
+                );
```
Comment (Nick97-git) [inline src/main/java/core/basesyntax/FigureSupplier.java:43]: create a separate method for each figure that will be returning a random figure, for example, getRandomSquare() and so on

**Review by Nick97-git — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/AreaCalculate.java [hunk 10]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface AreaCalculate {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/AreaCalculate.java:3]: Don’t use verbs for class/interface names.
`public interface AreaCalculator`

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 11]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private static Random random = new Random();
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: Do we need static here?

File: src/main/java/core/basesyntax/Figure.java [hunk 12]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable, AreaCalculate {
+    private final String color;
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Figure.java:4]: Do we need final here? What if u want change color?

File: src/main/java/core/basesyntax/Main.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figures = {
+                null, null, null,
+                new Square("cyan", 9),
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Main.java:7]: The other half of figures should have the same, default parameters.

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Circle.java [hunk 14]
```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private final int radius;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Circle.java:4]:
```suggestion
    private int radius;
```
there is no need to make this field final

File: src/main/java/core/basesyntax/Square.java [hunk 15]
```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private final int side;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Square.java:4]:
```suggestion
    private int side;
```

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 16]
```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private final int firstLeg;
+    private final int secondLeg;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/RightTriangle.java:5]:
```suggestion
    private int firstLeg;
    private int secondLeg;
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 17]
```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private final int longSide;
+    private final int shortSide;
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Rectangle.java:5]:
```suggestion
    private int longSide;
    private int shortSide;
```

**Review by aaananas — CHANGES_REQUESTED** [review]

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**